roxie/ccd/ccdserver.cpp

```diff
@@ -22601,18 +22688,21 @@ class CRoxieServerIndexActivity : public CRoxieServerActivity, implements IRoxie
         varFileInfo.setown(resolveLFN(indexName, isOpt));
         if (varFileInfo)
         {
-            translators.setown(new TranslatorArray) ;
-            keySet.setown(varFileInfo->getKeyArray(factory->queryActivityMeta(), translators, isOpt, isLocal ? factory->queryQueryFactory().queryChannel() : 0, getEnableFieldTranslation()));
+            unsigned formatCrc = getFormatCrc(indexHelper.getFormatCrc());
+            IOutputMetaData *projectedMeta = indexHelper.queryProjectedDiskRecordSize();
+            IOutputMetaData *expectedMeta = indexHelper.queryDiskRecordSize();
+            translators.setown(varFileInfo->getTranslators(formatCrc, projectedMeta, expectedMeta, getEnableFieldTranslation()));
```

Comment:
Does this need to remove the previous translators, otherwise if used in a loop or child query with dynamic filenames this list could get very long.

Reply:
I don't really understand the question. What list?

Reply:
translators array.  Each resolution will add an entry to the array.  I suspect the old code had the same problem

Reply:
translators.setown() is replacing previous translators each time. Still not sure I am understanding the issue you are suggesting?

Reply:
misreading the code...